common/utilities/timing.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import time
+from functools import wraps
+
+
+class StopWatch:
+
+    start_time = None
+
+    def start_timer(self):
+        self.start_time = time.time()
+
+    def stop_timer(self):
+        end_time = time.time()
+        diff = end_time - self.start_time
+        return diff
+
+
+def time_method(method):
```

Comment:
There's quite a lot of duplication between these methods (especially in the blocks that calculate the duration of the timer and print it to the log). I think it would be worth trying to extract some of this logic into separate reusable methods.

Reply:
I've pulled the duplicate pieces out into their own methods, maybe went a bit overkill but the individual timing methods are fairly concise now.